persistence/jdbc-common/src/main/java/org/infinispan/persistence/jdbc/common/configuration/AbstractJdbcStoreConfigurationSerializer.java

```diff
@@ -40,6 +40,12 @@ private void writeJDBCStoreConnection(ConfigurationWriter writer, ManagedConnect
       writer.writeEndElement();
    }
 
+   private void writeJDBCStoreConnection(ConfigurationWriter writer, CDIConnectionFactoryConfiguration configuration) {
+      writer.writeStartElement(Element.CDI_DATA_SOURCE);
+      configuration.attributes().write(writer);
+      writer.writeEndElement();
+   }
+
    protected void writeJDBCStoreConnection(ConfigurationWriter writer, AbstractJdbcStoreConfiguration configuration) {
```

Comment:
if you feel like refactor this class in this PR, I would suggest a generic method
```java
   protected void writeJDBCStoreConnection(ConfigurationWriter writer, AbstractJdbcStoreConfiguration configuration) {
      ConnectionFactoryConfiguration cfc = configuration.connectionFactory();
      if (cfc instanceof SimpleConnectionFactoryConfiguration) {
         writeAttributes(writer, Element.SIMPLE_CONNECTION, ((SimpleConnectionFactoryConfiguration) cfc).attributes());
      } else if (cfc instanceof PooledConnectionFactoryConfiguration) {
         writeAttributes(writer, Element.CONNECTION_POOL, ((PooledConnectionFactoryConfiguration) cfc).attributes());
      } else if (cfc instanceof ManagedConnectionFactoryConfiguration) {
         writeAttributes(writer, Element.DATA_SOURCE, ((ManagedConnectionFactoryConfiguration) cfc).attributes());
      } else if (cfc instanceof CDIConnectionFactoryConfiguration) {
         writeAttributes(writer, Element.CDI_DATA_SOURCE, ((CDIConnectionFactoryConfiguration) cfc).attributes());
      }
   }
   
   private void writeAttributes(ConfigurationWriter writer, Element element, AttributeSet attributes) {
      writer.writeStartElement(element);
      attributes.write(writer);
      writer.writeEndElement();
   }
```